### PR TITLE
Add product discount resource

### PR DESCRIPTION
**Pull request by onur9**

### CODE REVIEW

File: commercetools/resource_product_discount.go [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+		Value:       &value,
+		Predicate:   d.Get("predicate").(string),
+		SortOrder:   d.Get("sort_order").(string),
+		IsActive:    d.Get("is_active").(bool),
+	}
+
+	key := stringRef(d.Get("key"))
+	if *key != "" {
+		draft.Key = key
+	}
```
Comment (mvantellingen) [inline commercetools/resource_product_discount.go:167]: I've added in utils a `nilIEmpty()` function. so you can write `key := nilIfEmpty(stringRef(d.Get("key")))`

File: commercetools/resource_product_discount.go [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+	d.SetId(productDiscount.ID)
+	d.Set("version", productDiscount.Version)
+
+	return resourceProductDiscountRead(ctx, d, m)
+}
+
+func resourceProductDiscountRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] Reading product discount from commercetools, with productDiscount id: %s", d.Id())
```
Comment (mvantellingen) [inline commercetools/resource_product_discount.go:206]: Let's remove these statements, they don't add any value above running with `TF_LOG=trace CTP_DEBUG=1`

File: commercetools/resource_product_discount.go [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+	if err != nil {
+		if IsResourceNotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	if productDiscount == nil {
+		log.Print("[DEBUG] No product discount found")
```
Comment (mvantellingen) [inline commercetools/resource_product_discount.go:220]: Can be rmeoved

File: commercetools/resource_product_discount.go [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+		d.SetId("")
+	} else {
+		log.Print("[DEBUG] Found following product discount:")
+		log.Print(stringFormatObject(productDiscount))
+
```
Comment (mvantellingen) [inline commercetools/resource_product_discount.go:225]: Both lines can be removed

File: commercetools/resource_product_discount.go [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+	value, err := expandProductDiscountValue(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	draft := platform.ProductDiscountDraft{
+		Name:        name,
+		Description: &description,
+		Value:       &value,
+		Predicate:   d.Get("predicate").(string),
```
Comment (mvantellingen) [inline commercetools/resource_product_discount.go:159]: Should predicate be nil if empty?

File: commercetools/resource_product_discount.go [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+	value := d.Get("value").([]interface{})[0].(map[string]interface{})
+	switch value["type"].(string) {
+	case "relative":
+		return platform.ProductDiscountValueRelativeDraft{
+			Permyriad: value["permyriad"].(int),
+		}, nil
+	case "absolute":
+		// money := expandTypedMoney(value)
+		return platform.ProductDiscountValueAbsoluteDraft{
+			// Money: money,
```
Comment (mvantellingen) [inline commercetools/resource_product_discount.go:377]: This seems not finished

File: commercetools/resource_product_discount_test.go [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+					),
+					resource.TestCheckResourceAttr(
+						"commercetools_product_discount.standard", "value.0.type", "relative",
+					),
+					resource.TestCheckResourceAttr(
+						"commercetools_product_discount.standard", "value.0.permyriad", "1000",
+					),
+					resource.TestCheckResourceAttr(
+						"commercetools_product_discount.standard", "is_active", "true",
+					),
```
Comment (mvantellingen) [inline commercetools/resource_product_discount_test.go:51]: Let's add a check as in 
 - https://github.com/labd/terraform-provider-commercetools/pull/265/files#diff-f01e4c64f1877c4a0abc1852295c2046f5a73d2eaac80b17297147734a066aa3R340
 -  https://github.com/labd/terraform-provider-commercetools/pull/265/files#diff-f01e4c64f1877c4a0abc1852295c2046f5a73d2eaac80b17297147734a066aa3R164

File: commercetools/resource_product_discount.go (hunk 5 above)
Comment (onur9) [reply commercetools/resource_product_discount.go:160]: I believe it is required always

https://docs.commercetools.com/api/projects/productDiscounts

**Comment by mvantellingen** [comment]
> @mvantellingen
> 
> * `references` seem to be not needed (they are only computed, not inserted)
Yes, we can completely ignore this field for terraform (not even add it to the schema)

> * I couldn't set `fractionDigits` (absolute money type) from the API. The request seems to be ok, it must be the CT that blocks it from changing.
> 
> If we leave out `fraction_digits` for now, this is working fine actually.
Yes as just discussed on slack; it seems product discounts only work with CentPrecisionMoney and fraction digits are 'hardcoded' there. Let's ignore this field too.

**Comment by onur9** [comment]
@mvantellingen 

Removed fraction digits, but only keeping it as computed attribute.

I tried this API call:

```
curl -X POST https://api.europe-west1.gcp.commercetools.com/$1/product-types/$2" -i \
--header "Authorization: Bearer $3" \
--header 'Content-Type: application/json' \
--data-binary @- << DATA 
{

  "version": 1,
  "actions": [
  {
    "action" : "changeValue",
    "value" : {
      "type" : "relative",
      "permyriad" : 2000
    }
  }
  ]
}
DATA
```

It seems to return `401`, maybe it's the permission scope I missed. I'll check that later.

**Pull request merged**